### PR TITLE
phpPackages.phan: init at 5.4.1

### PR DESCRIPTION
**Pull request by apeschar**
###### Description of changes

Add the Phan static analyzer for PHP.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/php-packages/phan/default.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{
+  fetchurl,
+  lib,
+  makeWrapper,
+  mkDerivation,
+  php,
+}: let
```
Comment (viraptor) [inline pkgs/development/php-packages/phan/default.nix:7]:
```suggestion
{ fetchurl
, lib
, makeWrapper
, mkDerivation
, php
}:
let
```

File: pkgs/development/php-packages/phan/default.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  php' = php.withExtensions ({
+    enabled,
+    all,
+  }:
+    enabled ++ [all.ast]);
```
Comment (viraptor) [inline pkgs/development/php-packages/phan/default.nix:12]: That's not very long - maybe keep it on one line? I found the split + indenting a bit confusing.
```suggestion
  php' = php.withExtensions ({ enabled, all }: enabled ++ [all.ast]);
```

Comment (viraptor) [inline pkgs/development/php-packages/phan/default.nix:12]: That's not very long - maybe keep it on one line? I found the split + indenting a bit confusing.
```suggestion
  php' = php.withExtensions ({ enabled, all }: enabled ++ [all.ast]);
```

File: pkgs/development/php-packages/phan/default.nix (hunk 1 above)
Comment (jtojnar) [reply pkgs/development/php-packages/phan/default.nix:7]: I assume this is output of [Alejandra](https://github.com/kamadorueda/alejandra) formatter.

**Review by viraptor — APPROVED** [review]
works on darwin x86

**Pull request merged**